core/src/main/java/greencity/config/ScheduleConfig.java

```diff
@@ -11,5 +18,25 @@
  */
 @Configuration
 @EnableScheduling
+@AllArgsConstructor
 public class ScheduleConfig {
+    private final FactTranslationRepo factTranslationRepo;
+
+    /**
+     * Once a day randomly chooses new fact of day that has not been fact of day during this iteration.
+     * factOfDay == 0 - wasn't fact of day, 1 - is today's fact of day, 2 - already was fact of day.
+     */
+    @Transactional
+    @Scheduled(cron = "0 0 0 * * *")
+    public void chooseNewFactOfTheDay() {
+        Optional<List<FactTranslation>> list = factTranslationRepo.findRandomFact();
+        if (list.isPresent()) {
+            factTranslationRepo.updateFactOfDayStatus(1, 2);
+        } else {
+            factTranslationRepo.updateFactOfDayStatus(2, 0);
+            factTranslationRepo.updateFactOfDayStatus(1, 2);
+            list = factTranslationRepo.findRandomFact();
+        }
+        factTranslationRepo.updateFactOfDayStatusByHabitfactId(1, list.get().get(0).getHabitFact().getId());
```

Comment:
Please, move magic numbers to constants to make code more readable

Reply:
The best approach is to create Enum FactStatus or smth like this for such purposes